### PR TITLE
Implement SELECT *

### PR DESCRIPTION
**Pull request by krischerven**

### CODE REVIEW

File: memory.go [hunk 1]
```diff
@@ -243,6 +243,27 @@ func (mb *MemoryBackend) Select(slct *SelectStatement) (*Results, error) {
 		t.rows = [][]MemoryCell{{}}
 	}
 
+	// handle SELECT *
+	if len(*slct.item) == 1 && (*slct.item)[0].asterisk {
```
Comment (eatonphil) [inline memory.go:247]: I don't think this should or needs to be handled as a special case. I assume the behavior of using `*` is to display all columns that haven't already been a part of the select column statement. So I think we'd want some logic to look through all columns referenced and only add the rest.

Whatever the case, I don't think it's actually a special case?

Comment (krischerven) [reply memory.go:247]: I just checked and in postgres it adds all columns, including ones that are already a part of the SELECT statement

**Comment by krischerven** [comment]
The code is more flexible now, but there's a bug are a couple bugs. Will fix soon

**Comment by krischerven** [comment]
Seems to be all working now but I have to add tests.

**Comment by krischerven** [comment]
Actually, does this require testing @eatonphil? I don't a test file for memory.go. Code is working at any rate.

**Comment by eatonphil** [comment]
Yeah should probably start writing tests for the memory backend but since I haven't started that, no expectations there.

Tested and this works, thanks!

File: memory.go [hunk 2]
```diff
@@ -243,6 +243,33 @@ func (mb *MemoryBackend) Select(slct *SelectStatement) (*Results, error) {
...
+					asterisk: false,
+					as:       nil,
+				}
+				newItems = append(newItems, newSelectItem)
+			}
+			finalItems = append(finalItems, newItems...)
+		} else {
+			finalItems = append(finalItems, item)
+		}
+	}
```
Comment (eatonphil) [inline memory.go:271]: Interesting approach. Seems fine for now.

**Pull request merged**